shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/jdbc/datasource/factory/JDBCRawBackendDataSourceFactory.java

```diff
@@ -62,7 +63,17 @@ public DataSource build(final String dataSourceName, final DataSourceParameter d
         config.setDriverClassName(driverClassName);
         config.setJdbcUrl(dataSourceParameter.getUrl());
         config.setUsername(dataSourceParameter.getUsername());
-        config.setPassword(dataSourceParameter.getPassword());
+        String decryptPasswd = null;
+        try {
+            decryptPasswd = AlgorithmSecureFactory.getInstance()
+                    .decryptBackend(dataSourceParameter.getPassword());
+            // CHECKSTYLE:OFF
+        } catch (final Exception ex) {
+            // CHECKSTYLE:ON
+            log.error("password decrypt failed:", ex);
+            return null;
```

Comment:
What happen if null returned?

Reply:
this is no need repair,.
if password is null, proxy can't started. the log will record reason is password decrypt failed!